packages/server-wallet/src/onchain-service/transaction-submission.ts

```diff
@@ -0,0 +1,124 @@
+import {providers, Wallet} from 'ethers';
+import PriorityQueue from 'p-queue';
+
+import {
+  TransactionSubmissionServiceInterface,
+  MinimalTransaction,
+  TransactionSubmissionOptions,
+} from './types';
+
+// FIXME: replace with
+// import {WalletError as ChannelWalletError} from '@statechannels/server-wallet';
+import {WalletError as ChannelWalletError} from '..';
+
+type Values<E> = E[keyof E];
+
+export class TransactionSubmissionError extends ChannelWalletError {
+  readonly type = ChannelWalletError.errors.OnchainError;
+
+  // Errors where transactions will be safely retried
+  static readonly knownErrors = {
+    badNonce: `the tx doesn't have the correct nonce`,
+    invalidNonce: 'Invalid nonce',
+    noHash: 'no transaction hash found in tx response',
+    underpricedReplacement: 'replacement transaction underpriced',
+  };
+
+  static isKnownErr(errorMessage: string): boolean {
+    const idx = Object.values(TransactionSubmissionError.knownErrors).findIndex(err =>
+      errorMessage.includes(err)
+    );
+    return idx !== -1;
+  }
```

Comment:
I think I meant that this seems generic enough to be on `ChannelWalletError`

Reply:
Other wallet errors don't have the `knownErr`s fields, but can add it to the class. Is there some other instance in the wallet where this type of error message checking would be useful?